File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product = 1;
        int no_zero = 0;
        int lproduct[nums.size()];
        int rproduct[nums.size()];
        lproduct[0] = nums[0];
        rproduct[nums.size()-1] = nums[nums.size()-1];
        for(int i=1;i<nums.size();i++){
            lproduct[i] = lproduct[i-1]*nums[i];
        }
        for(int i=nums.size()-2;i>=0;i--){
            rproduct[i] = rproduct[i+1]*nums[i];
        }
        vector<int> ans;
        for(int i=0;i<nums.size();i++){
            if(i==0){
                // if(no_zero==1)
                ans.push_back(rproduct[i+1]);
                // else
            }
            else if(i==nums.size()-1){
                ans.push_back(lproduct[i-1]);

            }
            else{
                ans.push_back(rproduct[i+1]*lproduct[i-1]);
            }
        }
        return ans;

    }
};

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def swap(self,nums,i,j):
            a = nums[i]
            nums[i] = nums[j]
            nums[j] = a
            return
    def permute(self, nums: List[int]) -> List[List[int]]:
        ans = []

        
        def req(i):
            if i==len(nums):
                print(i)
                ans.append(nums.copy())
                return
        
            req(i+1)
            j = i+1

            while j<len(nums):
                self.swap(nums,i,j)
                req(i+1)
                self.swap(nums,i,j)
                j+=1
            return
        req(0)
        return ans
